packages/peregrine/lib/talons/Header/useCartTrigger.js

```diff
@@ -33,6 +33,7 @@ export const useCartTrigger = props => {
     const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
 
     const itemCount = data ? data.cart.total_quantity : 0;
+    const showCartTrigger = ['/checkout'].includes(history.location.pathname);
```

Comment:
I could have gone for a simple checkout page check instead of the array, but after my discussion with UX and @jimbo, I have realized this is can be a configurable option hence made an array to represent routes on which the cart trigger should be hidden.